pywbem_mock/_wbemconnection_mock.py

```diff
@@ -449,7 +450,117 @@ def add_cimobjects(self, objects, namespace=None):
+
+            method_callback (:term:`callable`):
+                The user-defined function or method that well be called when by
+                FakeWBEMConnection when the InvokeMethod for the defined
+                class and methodname is received.  This method will be called
+                with the parameters defined on the InvokeMethod call
+
+                The callback method defined by the user is expected to comply
+                with the following signature:
+
+                Parameters:
+
+                  conn (:class:`~pywbem.WBEMConnection`):
+                    Connection to the FakeWBEMConnection and its
+                    repository. This can be used to execute requests
+                    on the mock repository to manipulate objects in the
+                    repository (ex. cc = conn.GetClass(objectname)
+
+                  objectname:
+                    The object path of the target object, as follows:
+
+                    * Instance-level call: The instance path of the
+                      target instance, as a :class:`~pywbem.CIMInstanceName`
+                      object including the execution namespace (either the
+                      namespace defined by the client
+                      InvokeMethod or the WBEMConnection default namespace).
+
+                    * Class-level call: The class path of the target
+                      class, as a :class:`~pywbem.CIMClassName` object
+                      including the execution namespace (either the
+                      namespace defined by the client nvokeMethod or the
+                      WBEMConnection default namespace).
+
+                      The :class:`~pywbem.CIMClassName` object,  `host`
+                      attribute will be ignored.
+
+                  methodname(:term:`string`):
+                    The methodname from the InvokeMethod call. This allows
+                    a single callback function to be used as the responder
+                    for multiple methods.
+
+                  params (:term:`py:iterable` of tuples of name,value):
+                    Input parameters from the InvokeMethod Params input.
+
+                    An iterable of input parameters for the CIM method.
+
+                    Each iterated item represents a single input parameter
+                    for the CIM method and must be a ``tuple(name, value)``,
+                    with these tuple items:
+
+                    * name (:term:`string`):
+                      Parameter name (case independent)
+                    * value (:term:`CIM data type`):
+                      Parameter value
+
+                Return:
+                  * The user written callback method must return a tuple
+                    consisting of
+                      * ReturnValue (:term:`CIM data type`):
+                          Return value from the executed method.
+
+                      * outparams (`NocaseDict`_):
```

Comment:
So far, users cannot create `NocaseDict` objects because that symbol is not public. They only get `NocaseDict` objects generated by pywbem. This design however, requires that they be able to generate `NocaseDict` objects.

**Discussion Point:** Should we make `NocaseDict` public so that users can create objects of that type, or should we change the design e.g. to list(tuple(name,value))?

Reply:
We agreed to not make NocaseDict public and to redefine this interface and map to NocaseDict internally.